### PR TITLE
Malpedia

### PR DESCRIPTION
**Pull request by sebdraven**
add malware malpedia and add actors malpedia

### CODE REVIEW

File: plugins/feeds/public/malpedia_actors.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from typing import ClassVar
+
+from core import taskmanager
+from core.schemas import entity, task
+
+
+class Malpedia_Actors(task.FeedTask):
+    _defaults = {
+        "frequency": timedelta(days=1),
+        "name": "Malpedia",
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_actors.py:12]: Adjust name to account for type

File: plugins/feeds/public/malpedia_malware.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    def analyze_entry_malware(self, name_malware, entry: dict):
+        """Analyzes an entry as specified in the malpedia json."""
+        try:
+            m = entity.Malware(
+                name=entry["common_name"], description=entry["description"]
+            )
+        except Exception as e:
+            logging.error(f"Error creating malware entity: {e}")
+            return
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_malware.py:37]: Why catch the error here but noot in the previous one? What error are you expecting?

File: plugins/feeds/public/malpedia_malware.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        if entry.get("alt_names"):
+            m.aliases = entry["alt_names"]
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_malware.py:39]:
```suggestion
        m.aliases = entry.get("alt_names", [])
```

File: plugins/feeds/public/malpedia_malware.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        if entry.get("description"):
+            if m.description:
+                m.description += "## Malpedia\n\n"
+            else:
+                m.description = "## Malpedia \n\n"
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_malware.py:44]:
```suggestion
                m.description = "## Malpedia\n\n"
```

File: plugins/feeds/public/malpedia_malware.py (hunk 2 above)
Comment (tomchop) [reply plugins/feeds/public/malpedia_malware.py:37]: Same comment as above - you're not loading anything from the database here.

File: plugins/feeds/public/malpedia_malware.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            m.description += entry["description"]
+
+        if entry.get("urls"):
+            m.description += "\n\n"
+            m.description += "## External references\n\n"
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_malware.py:49]:
```suggestion
            m.description += "\n\n## External references\n\n"
```

File: plugins/feeds/public/malpedia_malware.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            for url in entry["urls"]:
+                m.description += f"* {url}\n"
+        if entry.get("type"):
+            m.family = entry["type"]
```
Comment (tomchop) [inline plugins/feeds/public/malpedia_malware.py:53]:
```suggestion
        m.family = entry.get("type", "")
```

File: plugins/feeds/public/malpedia_actors.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        tags = []
+
+        if intrusion_set.aliases:
+            tags += intrusion_set.aliases
+        tags.append(intrusion_set.name)
+        tags.append(name_actor)
+        try:
+            intrusion_set.tag(tags)
+        except Exception as e:
+            logging.error(f"Error tagging actor {name_actor}: {e}")
```
[the remark this answers is not in the document]
Comment (sebdraven) [reply plugins/feeds/public/malpedia_actors.py:55]: In malpedia you have tags malformed

File: plugins/feeds/public/malpedia_malware.py (hunk 2 above)
Comment (sebdraven) [reply plugins/feeds/public/malpedia_malware.py:37]: some entries has no name in malpedia

Comment (tomchop) [reply plugins/feeds/public/malpedia_malware.py]: In that case let's check the name and skip the creation attempt instead of try / catching

**Comment by tomchop** [comment]
We need to use the context feature from https://github.com/yeti-platform/yeti/pull/1089 for this, otherwise the description will get appended to the same entity every time the feed runs.

**Pull request merged**